### PR TITLE
Update duplicate definition

### LINKED ISSUE #60: Update duplicate person definition
Current definition is that any two people with identical names are the same person (duplicate).

MVP product definition requires that two people are only considered duplicates when both their names and phone numbers are identical.

### PR DESCRIPTION
**Pull request by Andrew22Teoh**
closes #60 

Tests have been updated, all look clear to me. Only change is in the test file and the class equality definition.
No documentation changes necessary, no suitable place to add it in either. Can leave it as something for the user to learn when they run into it.

### CODE REVIEW

File: src/test/java/seedu/address/model/person/PersonTest.java [hunk 1]
```diff
@@ -32,9 +32,14 @@ public void isSamePerson() {
...
 
-        // same name, all other attributes different -> returns true
+        // same name, all other attributes different -> returns false
         Person editedAlice = new PersonBuilder(ALICE).withPhone(VALID_PHONE_BOB).withEmail(VALID_EMAIL_BOB)
                 .withAddress(VALID_ADDRESS_BOB).withTags(VALID_TAG_HUSBAND).build();
+        assertFalse(ALICE.isSamePerson(editedAlice));
+
+        // same name, same phone number, all other attributes different -> returns true
+        editedAlice = new PersonBuilder(ALICE).withEmail(VALID_EMAIL_BOB).withAddress(VALID_ADDRESS_BOB)
+                .withTags(VALID_TAG_HUSBAND).build();
```
Comment (matchastack) [inline src/test/java/seedu/address/model/person/PersonTest.java]: nit: reuse of variable but for different logic. I think renaming to `aliceSameName` and `aliceSameNameAndNumber` might be better for clarity and maintainability.

**Review by matchastack — APPROVED** [review]
Just some nitpicking but LGTM!

File: src/test/java/seedu/address/model/person/PersonTest.java (hunk 1 above)
Comment (Andrew22Teoh) [reply src/test/java/seedu/address/model/person/PersonTest.java]: Good catch, I've edited this accordingly!

**Review by matchastack — APPROVED** [review]
LGTM!

**Pull request merged**